File: DSA/Anagram_check.cpp
#include<iostream>
using namespace std;

int main()
{
    string a = "listen";
    string b = "silent";

    int count1[26] = {0};
    int count2[26] = {0};

    for(int i=0; i<a.length(); i++)
    {
        count1[a[i]-'a']++;
    }

    for(int i=0; i<b.length(); i++)
    {
        count2[b[i]-'a']++;
    }

    bool isAnagram = true;

    for(int i=0; i<26; i++)
    {
        if(count1[i] != count2[i])
        {
            isAnagram = false;
            break;
        }
    }

    if(isAnagram)
        cout<<"Anagram";
    else
        cout<<"Not Anagram";

    return 0;
}

File: DSA/Anagram_check.py
a = "listen"
b = "silent"

count1 = [0] * 26
count2 = [0] * 26

# pehle word ka count
for i in range(len(a)):
    count1[ord(a[i]) - ord('a')] += 1

# dusre word ka count
for i in range(len(b)):
    count2[ord(b[i]) - ord('a')] += 1


isAnagram = True

# dono counts compare karna
for i in range(26):
    if count1[i] != count2[i]:
        isAnagram = False
        break


if isAnagram:
    print("Anagram")
else:
    print("Not Anagram")